### PR TITLE
libxls: add recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **libxls/1.6.2**

libxls is xls(old type Excel) file reader library.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — APPROVED** [review]

**Comment by uilianries** [comment]
@toge the error is not clear, but it seems it occurred as a new windows when building, that's why no error appeared in the log. I would recommend you skipping `MT` build with https://docs.conan.io/en/latest/reference/conanfile/tools/microsoft.html#conan-tools-microsoft-is-msvc-static-runtime

**Comment by uilianries** [comment]
Toge, please, consider https://github.com/toge/conan-center-index/pull/3

**Comment by toge** [comment]
@uilianries 
Thank a lot!
I try to use autotools.

**Comment by uilianries** [comment]
Need further investigation. It does not fail for me, but I use Mac M1. C3I is cross-building x86_64 to Arm64 on Mac.

**Comment by toge** [comment]
I'm trying to solve it, too.
It's funny there is the one OK condition.

**Comment by uilianries** [comment]
@toge Yes, we have few good examples, but they work, as this one:https://github.com/conan-io/conan-center-index/pull/13263/files

Basically, you need to get `Environment` object from AutotoolsToochains, set `CC=cl.exe` and others.

Another good option and more clear (Which I prefer): https://github.com/conan-io/conan-center-index/blob/master/recipes/libbacktrace/all/conanfile.py#L91

In case it does not work, please, skip it.

**Comment by toge** [comment]
@uilianries 
Thanks a lot!
I try them.

**Comment by toge** [comment]
I have given up support for Windows...

File: recipes/libxls/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
+from audioop import cross
```
Comment (prince-chrismc) [inline recipes/libxls/all/conanfile.py:1]: Linter

```suggestion
```

File: recipes/libxls/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.gnu import Autotools, AutotoolsToolchain, AutotoolsDeps
+from conan.tools.layout import basic_layout
+from conan.tools.files import export_conandata_patches, apply_conandata_patches, rmdir, copy, save, replace_in_file, get, rm
```
Comment (prince-chrismc) [inline recipes/libxls/all/conanfile.py:6]: Linter

```suggestion
from conan.tools.files import export_conandata_patches, apply_conandata_patches, rmdir, copy, save, get, rm
```

File: recipes/libxls/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+from conan.tools.apple import is_apple_os
+from conan.tools.build import cross_building
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.scm import Version
```
Comment (prince-chrismc) [inline recipes/libxls/all/conanfile.py:10]:
```suggestion
```

File: recipes/libxls/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    def layout(self):
+        basic_layout(self, src_folder='src')
+
+    def requirements(self):
+        if not is_apple_os(self):
+            self.requires("libiconv/1.17")
+
+    def validate(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration(f"{self.ref} doesn't support Windows(yet).")
```
Comment (prince-chrismc) [inline recipes/libxls/all/conanfile.py:57]:
```suggestion
            raise ConanInvalidConfiguration(f"{self.ref} doesn't support Windows (yet). Contributions are always welcomed")
```

File: recipes/libxls/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        autotools.install()
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rm(self, "*.la", os.path.join(self.package_folder, "lib"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["xlsreader"]
+
+        if is_apple_os(self):
+            self.cpp_info.system_libs.append("iconv")
```
Comment (prince-chrismc) [inline recipes/libxls/all/conanfile.py:105]: I am surprised it's a system lib on Mac 🤔 not usually the case... care to share with me? I am not familar with this project

```suggestion
        if is_apple_os(self):
            self.cpp_info.system_libs.append("iconv")
        else:
            self.cpp_info.requires.append("libiconv::libiconv")
```

File: recipes/libxls/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+        self.cpp_info.set_property("cmake_file_name", "libxls")
+        self.cpp_info.set_property("cmake_target_name", "libxls::libxls")
+        self.cpp_info.set_property("pkg_config_name", "libxls")
+
+        if not is_apple_os(self):
+            self.cpp_info.requires.append("libiconv::libiconv")
+
+        # TODO: Remove in Conan 2.0
```
Comment (prince-chrismc) [inline recipes/libxls/all/conanfile.py:114]:
```suggestion
        # TODO: Remove in Conan 2.0
```

**Review by prince-chrismc — COMMENTED** [review]
As always your work is out standinging.

I just had one question and a few linter things I pointed out :)

File: recipes/libxls/all/conanfile.py (hunk 5 above)
Comment (toge) [reply recipes/libxls/all/conanfile.py:102]: @prince-chrismc 
Sorry, I neglected it so much that I forgot some of it.
I will investigate again.

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
CI is happy and this a a lot of great work, I'd love to see it go in

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**